File: common/src/main/java/juuxel/adorn/loot/AdornLootFunctionTypes.java
package juuxel.adorn.loot;

import juuxel.adorn.lib.registry.Registered;
import juuxel.adorn.lib.registry.Registrar;
import juuxel.adorn.lib.registry.RegistrarFactory;
import net.minecraft.world.level.storage.loot.functions.LootItemFunctionType;
import net.minecraft.core.registries.Registries;

public final class AdornLootFunctionTypes {
    public static final Registrar<LootItemFunctionType<?>> LOOT_FUNCTION_TYPES = RegistrarFactory.get().create(Registries.LOOT_FUNCTION_TYPE);
    public static final Registered<LootItemFunctionType<CheckTradingStationOwnerLootFunction>> CHECK_TRADING_STATION_OWNER =
        LOOT_FUNCTION_TYPES.register("check_trading_station_owner", () -> new LootItemFunctionType<>(CheckTradingStationOwnerLootFunction.CODEC));

    public static void init() {
    }
}

File: common/src/main/kotlin/juuxel/adorn/loot/AdornLootFunctionTypes.kt
package juuxel.adorn.loot

import juuxel.adorn.lib.registry.Registrar
import juuxel.adorn.lib.registry.RegistrarFactory
import net.minecraft.loot.function.LootFunctionType
import net.minecraft.registry.RegistryKeys

object AdornLootFunctionTypes {
    val LOOT_FUNCTION_TYPES: Registrar<LootFunctionType> = RegistrarFactory.get().create(RegistryKeys.LOOT_FUNCTION_TYPE)
    val CHECK_TRADING_STATION_OWNER: LootFunctionType by LOOT_FUNCTION_TYPES.register("check_trading_station_owner") {
        LootFunctionType(CheckTradingStationOwnerLootFunction.CODEC)
    }

    fun init() {
    }
}
